docs/modules/pipelines/pages/job-monitoring.adoc

```diff
@@ -0,0 +1,61 @@
+= Monitoring Jobs
+:description: To ensure the reliability of streaming jobs and their workloads, try implementing some basic monitoring. With Hazelcast, you can monitor the status of streaming jobs to quickly identify job failures for troubleshooting.
+
+{description}
+
+== Monitoring for Status Changes
+
+There are two ways to monitor for job status changes. You can either poll the `Job.getStatus()` method separately for each job or add a `JobStatusListener` to your jobs. In both cases, you can write custom code for execution on status change. For example, triggering an alert or writing the changed status to a dashboard. 
+
+== Using a JobStatusListener
+
+The `JobStatusListener` has several advantages over polling for the job status. 
+
+- The listener is notified immediately when a status change occurs. To achieve a similar result with polling, you might increase the polling frequency which would decrease performance.  
+
+- You can identify jobs in an error state. By default, xref:troubleshoot:error-handling.adoc#processing-guarantees[all streaming jobs are suspended on failure]. When a job moves to the suspended status, the listener is passed a suspension cause; either an error description or requested by a user. 
+
+NOTE: The reasons for job suspensions and failures are also displayed on the Jobs page in Management Center.
+
+The `JobStatusListener` also has a limitation. The listener may be attached to the job after its status changes to `RUNNING`. To cope with this situation, add the following code to your listener.
```

Comment:
This is incorrect. The listener can be attached to the job at any time after submission. However, the attachment is unsynchronized with the job execution, which will result in one of the following when listening to RUNNING status.
1. The attachment is quick enough, so that the RUNNING status will be caught.
2. The attachment happens after the RUNNING status is set, which will cause the listener to miss the (initial) notification.

Reply:
Statement has been updated. Could you supply a code snippet for `busy-wait`?